### DISCUSSION TITLE
How to build on M1 Mac?

### DISCUSSION THREAD

**Discussion by janhoy** [UCLALibrary/docker-cantaloupe#149] [General]
I'm trying to build a docker image with custom edits on my M1 Mac. Getting errors

> could not get native definition for type `POINTER`, original error message follows: java.lang.UnsatisfiedLinkError: Unable to execute or load jffi binary stub from `/var/folders/hb/_6h0qb492dzg72h1n0gw7fnw0000gn/T/`. Set `TMPDIR` or Java property `java.io.tmpdir` to a read/write path that is not mounted "noexec".

I think it is because JRuby version does not support M1 Mac, so I filed https://github.com/cantaloupe-project/cantaloupe/issues/628

But is there a way to build the Docker image on my M1 mac without upgrading JRuby/JFFI? I think the build crashes early, so it would not help patching cantaloupe's pom.xml file for this?

Guidance appreciated!

**Comment by ksclarke** [comment]
I don't have an M1 Mac to test with, but I've been thinking about requesting one (though I'm a full-time Linux user, I have an older work-assigned intel-based Mac for testing Mac compatibility so I've been thinking it makes sense to upgrade it to an M1 now).

Maybe someone else here has an M1 Mac and might have some insight (or a PR)? Your cantaloupe ticket about updating JRuby looks like a reasonable path to pursue to me.

**Reply by janhoy** [reply]
I filed #151 which helped me build multi platform docker images on my M1. It's at least one piece of the puzzle.

**Reply by ksclarke** [reply]
Great, thanks! I have a Mac now I can test on. Expect to get to this next week.
